components/org.wso2.carbon.identity.api.server.application.management/org.wso2.carbon.identity.api.server.application.management.v1/src/main/java/org/wso2/carbon/identity/api/server/application/management/v1/core/LoginFlowAIService.java

```diff
@@ -0,0 +1,240 @@
+
+        try {
+            List<Map<String, Object>> userClaims = loginFlowGenerateRequest.getUserClaims();
+            JSONArray userClaimsJsonArray = new JSONArray();
+            for (Map<String, Object> userClaim : userClaims) {
+                userClaimsJsonArray.put(new JSONObject()
+                        .put(CLAIM_URI_KEY, userClaim.get(CLAIM_URI_KEY))
+                        .put(DESCRIPTION_KEY, userClaim.get(DESCRIPTION_KEY)));
+            }
+
+            JSONObject availableAuthenticators = new JSONObject();
+            Map<String, List<Map<String, Object>>> authenticators = loginFlowGenerateRequest
+                    .getAvailableAuthenticators();
+            for (Map.Entry<String, List<Map<String, Object>>> entry : authenticators.entrySet()) {
+                String authenticatorType = entry.getKey();
+                List<Map<String, Object>> authenticatorList = entry.getValue();
+
+                JSONArray authenticatorsJsonArray = new JSONArray();
+                for (Map<String, Object> authenticator : authenticatorList) {
+                    authenticatorsJsonArray.put(new JSONObject()
+                            .put(AUTHENTICATOR_NAME_KEY, authenticator.get(AUTHENTICATOR_NAME_KEY))
+                            .put(AUTHENTICATOR_IDP_KEY, authenticator.get(AUTHENTICATOR_IDP_KEY))
+                            .put(DESCRIPTION_KEY, authenticator.get(DESCRIPTION_KEY)));
+                }
+                availableAuthenticators.put(authenticatorType, authenticatorsJsonArray);
+            }
+
+            String operationId = ApplicationManagementServiceHolder.getLoginFlowAIManagementService()
+                    .generateAuthenticationSequence(loginFlowGenerateRequest.getUserQuery(), userClaimsJsonArray,
+                            availableAuthenticators);
+            LoginFlowGenerateResponse response = new LoginFlowGenerateResponse();
+            response.setOperationId(operationId);
+            return response;
+        } catch (AIServerException e) {
+            throw handleServerException(e);
+        } catch (AIClientException e) {
+            throw handleClientException(e);
+        }
+    }
+
+    /**
+     * Get login flow AI generation result.
+     *
+     * @param operationId Operation ID of the login flow AI generation.
+     * @return LoginFlowResultResponse.
+     */
+    public LoginFlowResultResponse getLoginFlowAIGenerationResult(String operationId) {
+
+        try {
+            Object generationResult = ApplicationManagementServiceHolder.getLoginFlowAIManagementService()
+                    .getAuthenticationSequenceGenerationResult(operationId);
+            LoginFlowResultResponse response = new LoginFlowResultResponse();
+            Map<String, Object> generationResultMap = (Map<String, Object>) generationResult;
+            response.setStatus(getStatusFromResult(generationResultMap));
+            if (!((Map<?, ?>) generationResult).containsKey(AI_RESPONSE_DATA_KEY)) {
```

Comment:
instead of casting generationResult here use the already casted and created object generationResultMap